packages/components/src/Select/docs/selectMenu/dynamicLists.tsx

```diff
@@ -0,0 +1,71 @@
+import { Header, Inline, Select, Section, Collection } from "@hopper-ui/components";
+
+interface ListItemProps {
+    id: number | string;
+    name: string;
+}
+
+interface ListSectionProps {
+    name: string;
+    children: ListItemProps[];
+}
+
+export default function Example() {
+    const options = [
+        { id: 2, name: "Fred" },
+        { id: 3, name: "Bob" },
+        { id: 4, name: "Gabriel" },
+        { id: 6, name: "Sarah" },
+        { id: 7, name: "Louise" },
+        { id: 8, name: "Karen" }
+    ] satisfies ListItemProps[];
+
+    const optionsWithSections = [
+        {
+            name: "Boy Names", children: [
+                { id: 2, name: "Fred" },
+                { id: 3, name: "Bob" },
+                { id: 4, name: "Gabriel" }
+            ]
+        },
+        {
+            name: "Girl Names", children: [
+                { id: 6, name: "Sarah" },
+                { id: 7, name: "Louise" },
+                { id: 8, name: "Karen" }
+            ]
+        }
+    ] satisfies ListSectionProps[];
+
+    return (
+        <Inline alignY="flex-start">
+            <Select
+                aria-label="Names"
+                items={options}
+            >
+                {item => {
+                    const listItem = item as ListItemProps;
```

Comment:
Same comment here, I would type the item instead of casting (if possible).

But deconstruct. I will only comment here, but you can do this in a couple places in this PR.